File: app/src/main/java/com/github/rooneyandshadows/lightbulb/easyrecyclerviewdemo/demo/models/StickySimpleDemoModel.java
package com.github.rooneyandshadows.lightbulb.easyrecyclerviewdemo.demo.models;

import android.os.Parcel;

import com.github.rooneyandshadows.lightbulb.commons.utils.ParcelableUtils;
import com.github.rooneyandshadows.lightbulb.recycleradapters.abstraction.EasyAdapterDataModel;

public class StickySimpleDemoModel extends EasyAdapterDataModel {
    private final boolean isHeader;
    private final String title;
    private final String subtitle;


    public StickySimpleDemoModel(boolean isHeader, String title, String subtitle) {
        super(false);
        this.isHeader = isHeader;
        this.title = title;
        this.subtitle = subtitle;
    }

    // Parcelling part
    public StickySimpleDemoModel(Parcel in) {
        super(in);
        this.isHeader = ParcelableUtils.readBoolean(in);
        this.title = ParcelableUtils.readString(in);
        this.subtitle = ParcelableUtils.readString(in);
    }

    public static final Creator<StickySimpleDemoModel> CREATOR = new Creator<StickySimpleDemoModel>() {
        public StickySimpleDemoModel createFromParcel(Parcel in) {
            return new StickySimpleDemoModel(in);
        }

        public StickySimpleDemoModel[] newArray(int size) {
            return new StickySimpleDemoModel[size];
        }
    };

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        super.writeToParcel(parcel, i);
        ParcelableUtils
                .writeBoolean(parcel, isHeader)
                .writeString(parcel, title)
                .writeString(parcel, subtitle);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public String getItemName() {
        return title;
    }

    public boolean isHeader() {
        return isHeader;
    }

    public String getTitle() {
        return title;
    }

    public String getSubtitle() {
        return subtitle;
    }
}

File: app/src/main/java/com/github/rooneyandshadowss/lightbulb/easyrecyclerviewdemo/demo/models/StickySimpleDemoModel.kt
package com.github.rooneyandshadowss.lightbulb.easyrecyclerviewdemo.demo.models

import com.github.rooneyandshadows.lightbulb.commons.utils.ParcelUtils.Companion.writeString
import android.os.Parcel
import android.os.Parcelable.Creator
import com.github.rooneyandshadows.lightbulb.commons.utils.ParcelUtils
import com.github.rooneyandshadows.lightbulb.recycleradapters.abstraction.EasyAdapterDataModel

class StickySimpleDemoModel : EasyAdapterDataModel {
    val isHeader: Boolean
    val subtitle: String
    override val itemName: String

    constructor(isHeader: Boolean, title: String, subtitle: String) {
        this.isHeader = isHeader
        itemName = title
        this.subtitle = subtitle
    }

    // Parcelling part
    constructor(parcel: Parcel) {
        isHeader = ParcelUtils.readBoolean(parcel)!!
        itemName = ParcelUtils.readString(parcel)!!
        subtitle = ParcelUtils.readString(parcel)!!
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        writeString(parcel, subtitle)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Creator<StickySimpleDemoModel> {
        override fun createFromParcel(parcel: Parcel): StickySimpleDemoModel {
            return StickySimpleDemoModel(parcel)
        }

        override fun newArray(size: Int): Array<StickySimpleDemoModel?> {
            return arrayOfNulls(size)
        }
    }
}